spotbugs/src/main/java/edu/umd/cs/findbugs/filter/SourceMatcher.java

```diff
@@ -47,15 +47,25 @@ public SourceMatcher(String fileName) {
 
     @Override
     public boolean match(BugInstance bugInstance) {
+        // Match on the pure filename first
         ClassAnnotation primaryClassAnnotation = bugInstance.getPrimaryClass();
         if(primaryClassAnnotation == null){
             return false;
         }
+        
+        boolean result = false;
         String bugFileName = primaryClassAnnotation.getSourceFileName();
-        if(bugFileName == null || bugFileName.isEmpty()){
-            return false;
+        if(bugFileName != null && !bugFileName.isEmpty()){
+            result = fileName.match(bugFileName);
+            
+            // if no result try to match with the full path as well
+            if (!result && bugInstance.getPrimarySourceLineAnnotation().isSourceFileKnown()) {
+                bugFileName = bugInstance.getPrimarySourceLineAnnotation().getRealSourcePath();
+                result = fileName.match(bugFileName);
```

Comment:
Maybe we can omit this line? We overwrite the result at the 68th line.

Reply:
Will have a look to it.

Reply:
I removed the one from line #68 because of the following points:
1. the result is false by initialization. Only in case we have a valid bugfile name it will change.
2. than the result is written. in case it is negative we proceed checking with the full path of the file
3. finally the result is written again (overwritten).

If we remove the one in #64 and leave #68 we do an unneccesary check.

So i think, if there is no bug file name the result should be always false.

I updated the comments in the code, to help understand the logic.